src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/ApplicationValidationService.kt

```diff
@@ -37,63 +47,90 @@ class ApplicationValidationService(
-      LOG.error("Validation failed for application reference - ${application.reference} with msg - ${ve.message}")
-      throw ve
+      SYSTEM -> isSystemApplicationValid()
+    }
+
+    if (errorCodes.isNotEmpty()) {
+      throw ApplicationValidationException(errorCodes.toTypedArray())
     }
   }
 
   private fun isPublicApplicationValid(
     bookingRequestDto: BookingRequestDto?,
     application: Application,
     existingBooking: Visit?,
-  ) {
+  ): List<ApplicationValidationErrorCodes> {
+    val errorCodes = mutableListOf<ApplicationValidationErrorCodes>()
     val prison = application.prison
-    val prisoner = prisonerService.getPrisoner(application.prisonerId) ?: throw ValidationException("prisoner not found")
+    val prisoner = prisonerService.getPrisoner(application.prisonerId) ?: run {
+      LOG.info("Prisoner with id - ${application.prisonerId} not found.")
+      throw ApplicationValidationException(
+        APPLICATION_INVALID_PRISONER_NOT_FOUND,
+      )
+    }
 
     checkPrison(prison.code, prisoner.prisonCode)
-    checkSessionSlot(application, prisoner, prison)
+    checkSessionSlot(application, prisoner, prison)?.also {
+      errorCodes.add(it)
+    }
 
     // check if there are non-association visits that have been booked in after the application was created
     checkNonAssociationVisits(
       prisonerId = application.prisonerId,
       sessionDate = application.sessionSlot.slotDate,
       prisonId = application.prisonId,
-    )
+    )?.also {
+      errorCodes.add(it)
+    }
 
     // check if any double bookings for the same prisoner
-    checkDoubleBookedVisits(prisonerId = application.prisonerId, sessionSlot = application.sessionSlot)
+    checkDoubleBookedVisits(prisonerId = application.prisonerId, sessionSlot = application.sessionSlot, visitReference = application.visit?.reference)?.also {
+      errorCodes.add(it)
+    }
 
     // check prisoner's VOs - only applicable if user type = PUBLIC as staff can override VO count
-    checkVOLimits(application.prisonerId)
+    checkVOLimits(application.prisonerId)?.also {
+      errorCodes.add(it)
+    }
 
     // check capacity for slot
-    checkSlotCapacity(bookingRequestDto, application, existingBooking)
+    checkSlotCapacity(bookingRequestDto, application, existingBooking)?.also {
+      errorCodes.add(it)
+    }
+
+    return errorCodes.toList()
   }
 
   private fun isStaffApplicationValid(
     bookingRequestDto: BookingRequestDto?,
     application: Application,
     existingBooking: Visit?,
-  ) {
+  ): List<ApplicationValidationErrorCodes> {
+    val errorCodes = mutableListOf<ApplicationValidationErrorCodes>()
     // check capacity for slot
-    checkSlotCapacity(bookingRequestDto, application, existingBooking)
+    checkSlotCapacity(bookingRequestDto, application, existingBooking)?.also {
+      errorCodes.add(it)
+    }
+
+    return errorCodes
+  }
+
+  private fun isSystemApplicationValid(): List<ApplicationValidationErrorCodes> {
```

Comment:
not sure about using the is start to a method as this makes me think it will return a boolean
```
isStaffApplicationValid
isSystemApplicationValid
isPublicApplicationValid
```
May be use getStaffValidationErrors

Reply:
updated